packages/ws-worker/test/channels/worker-queue.test.ts

```diff
@@ -106,5 +106,61 @@ test('should fail to connect with an invalid auth token', async (t) => {
   });
 });
 
+test('should pass capacity in join payload when provided', (t) => {
+  return new Promise((done) => {
+    function createSocket(endpoint: string, _options: any) {
+      const socket = new MockSocket(endpoint, {}, async () => {});
+
+      // Override channel method to capture join params
```

Comment:
As mentioned above, I don't think this is a very productive pattern and I'd quite like to simplify